### PR TITLE
feat(FX-4637): User adds artworks to new list — initial artwork grid and selection

### PR DESCRIPTION
**Pull request by nickskalkin**
The type of this PR is: **Feat**

This PR solves [FX-4637]

### Description

The entire "User adds artworks to a new list" story is split in separate tickets, that's why I want to write down what is in scope of this ticket:

- Logic to open a new modal
- Modal header & sticky footer (only markup, no logic to count selected artworks)
- Hardcoded list of artworks
- 3 columns grid for desktop, 2 columns for mobile web 

What I left outside:

- Artwork metadata - will be done [FX-6437](https://artsyproduct.atlassian.net/browse/FX-4637)
- Tests - I will cover new logic with tests also within [FX-6437](https://artsyproduct.atlassian.net/browse/FX-4637)

# Demo

- [Review app](https://create-new-list-with-artworks.artsy.net/collector-profile/saves2)

| Desktop | mWeb |
|---|---|
| <video src="https://user-images.githubusercontent.com/3934579/223154615-0f99a3dd-3acc-4b72-b2dc-b532e7f20018.mp4" /> |  <video src="https://user-images.githubusercontent.com/3934579/223154774-50df78b6-7eb1-4a14-b7b7-cc4e8be77641.mp4" /> |

[FX-4637]: https://artsyproduct.atlassian.net/browse/FX-4637?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/CollectorProfile/Routes/Saves2/Components/SavesHeader.tsx
Comment (nickskalkin) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/SavesHeader.tsx:5]: Going to move this file up one directory (all create list-related files will be located in `CreateNewList` directory). I was waiting until your other PR got merged @dimatretyak

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal.tsx
Comment (nickskalkin) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal.tsx:27]: Changes is this file will vanish once I fix merge conflicts with https://github.com/artsy/force/pull/12021

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx
Comment (nickskalkin) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx:33]: I still couldn't find a way to split title in two lines. @dimatretyak I am thinking to allow passing a component instead of a string as a title, like header and footer do. What do you think about it?

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx:36]: @gkartalis as far as I remember, is it always better to avoid using `data-testid`? is that right?

Comment (gkartalis) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx:36]: depends 🤷  

I know that there are the following references in force:
- `data-test-id`
- `data-testid` 
- `data-test`

The integrity [docs](https://github.com/artsy/integrity#selecting-specific-elements) suggest to use `data-test` for e2e tests (I am guessing something similar should be the case for acceptance tests in force).

Also [`@testing-library/react`](https://testing-library.com/docs/queries/bytestid/) uses `data-testid` for the `byTestId` query.

So that makes me think that we should avoid the `data-test-id` 🤔

Comment (nickskalkin) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx:36]: gotcha, thanks!

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/SelectArtworkItem.tsx
Comment (dimatretyak) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/SelectArtworkItem.tsx:15]: and it is better to add the left offset at the place where the component is used (this will give us more flexibility if we want to use this component elsewhere. e.g [here](https://github.com/artsy/force/blob/ce8c2824423ec008e98cdffc93e55cfebb702d8b/src/Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListItem.tsx#L56-L60))

```javascript
<Box ml="1">
  <SelectArtworkItem isSelected={isSelected} />
</Box>
```

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx
Comment (dimatretyak) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx:37]: and maybe it would be better to use `alignItems="center"`?

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx:60]: **nitpick**: pass `selected="-position"`

Comment (dimatretyak) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx:60]: or better

```javascript
const SORTS = [{ text: "Recently Saved", value: "-position" }]
...
<SortFilter
  sortOptions={SORTS}
  selected={SORTS[0].value}
  onSort={() => {}}
/>
```

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx:25]: header and footer were supposed to be sticky, but for some reason they scroll together with the content

https://user-images.githubusercontent.com/3513494/223394860-a985b85c-ed5a-4543-a07f-0de2fd76ea4d.mp4

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx:28]: it's better to replace `width` on `minWidth`

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx:38]: we should use `sm-display` variant here

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx
Comment (dimatretyak) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx:33]: unfortunately, we will not be able split title in two lines with current implementation (we need to add changes to the `palette`). Maybe it's better to discuss this point with Ryan and use only one line (e.g. just `Add saved works to this list`)?

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx
Comment (nickskalkin) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModal.tsx:30]: hey @dimatretyak with `minWidth` my modal takes the whole width of the screen, so I left it like that

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworkItem.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworkItem.tsx:23]: **question**: do we plan use `role="option"` in the future?

Comment (nickskalkin) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworkItem.tsx:23]: Not sure, forgot to delete it, will do now!

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworkItem.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworkItem.tsx:24]: please replace on `flexDirection="column"`

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworkItem.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworkItem.tsx:27]: `justifyContent="space-between"`

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworkItem.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworkItem.tsx:10]: please remove `React` import above and just use `FC<ArtworkItemProps>` here

Comment (dimatretyak) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworkItem.tsx:10]: like [here](https://github.com/artsy/force/blob/e364c42066e4f296bc5a922faa247381f5ee6399/src/Apps/Article/ArticleApp.tsx#L20)

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworkItem.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworkItem.tsx:22]: **suggestion**: move `Column` to the `ArtworksList` component. Something like this

```javascript
<GridColumns>
  {TEMP_ARTWORKS.map(artwork => {
    return (
      <Column span={[6, 4]}>
        <ArtworkItem item={artwork} onItemPress={handleItemPress} />
      </Column>
    )
  })}
</GridColumns>
```

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModalWizard.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModalWizard.tsx:35]: we can skip react fragment (`<></>`) here

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModalWizard.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModalWizard.tsx:36]: **suggestion**: we can use `AddArtworksModal` here

Comment (dimatretyak) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModalWizard.tsx:36]: something like this

```javascript
if (listName !== null) {
  return (
    <AddArtworksModal
      onClose={onComplete}
      onComplete={handleAddArtworksComplete}
      listName={listName}
    />
  )
}

return (
  <CreateNewListModal
    onClose={onClose}
    onComplete={handleCreateListComplete}
  />
)
```

File: src/Apps/CollectorProfile/Routes/Saves2/Components/SavesHeader.tsx
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/SavesHeader.tsx:5]: suggestion: `import {...} from "./CreateNewListModal/CreateNewListModalWizard"`

**Review by dimatretyak — APPROVED** [review]
Thank you for all changes 🙏. Feel free to merge when you want

**Pull request merged**
